lib/src/main/java/com/cloudbees/groovy/cps/CpsTransformer.java

```diff
@@ -763,6 +763,41 @@ public void run() {
         BINARY_OP_TO_BUILDER_METHOD.put(KEYWORD_IN, "isCase");
     }
 
+    private void multipleAssignment(final Expression parentExpression,
+                                    final TupleExpression tuple,
+                                    final Expression rhs) {
+        List<Expression> tupleExpressions = tuple.getExpressions();
+
+        for (int i = 0, tupleExpressionsSize = tupleExpressions.size(); i < tupleExpressionsSize; i++) {
+            final Expression tupleExpression = tupleExpressions.get(i);
+            final int index = i;
+            // def (a, b, c) = [1, 2] is allowed - c will just be null in that scenario.
+            // def (a, b) = [1, 2, 3] is allowed as well - 3 is just discarded.
+            // def (a, b) = 4 will error due to Integer.getAt(int) not being a thing
+            // def (a, b) = "what" is allowed - a will equal 'w', and b will equal 'h'
+            makeNode("assign", new Runnable() {
+                @Override
+                public void run() {
+                    loc(parentExpression);
+                    visit(tupleExpression);
+                    makeNode("array", new Runnable() {
```

Comment:
So this might be able to be simpler, but it's my attempt to approximate https://github.com/apache/groovy/blob/3a035b97712f1e59c792a393c2ea4ec3f32a5871/src/main/org/codehaus/groovy/classgen/asm/BinaryExpressionHelper.java#L445-L464